methods_add_cards.py

```diff
@@ -0,0 +1,265 @@
+
+    Raises:
+        ValueError: informs the user of the disconnect between config file and the Project Card
+    """
+    if subject_id not in range_in_use:
+        msg = (
+            "New {} id ({}) in project '{}' is not in the base networks allowable range"
+            "({} to {}) as defined in the configuration file.".format(
+                nodes_or_links,
+                project_name,
+                min(range_in_use.keys()),
+                max(range_in_use.keys()),
+            )
+        )
+        raise ValueError(msg)
+
+
+def _is_id_used_in_base_network(
+    nodes_or_links: str,
+    project_name: str,
+    subject_id: int,
+    range_in_use: dict,
+):
+    """
+    Checks if new node or link id is used in the base network as defined in the config file
+
+    Args:
+        nodes_or_links (str): "node" or "link", which is used in error message
+        project_name (str): project name, which is used in error message
+        subject_id (int): the proposed new node or link id number
+        range_in_use (dict): a dictionary defining the id range with a bool indicating if the id number is used in the base network
+
+    Raises:
+        ValueError: informs the user of the disconnect between the config file and the Project Card
+    """
+    if subject_id in range_in_use:
+        if range_in_use[subject_id] == True:
+            msg = (
+                "New {} id ({}) in project '{}' is in use in the base network. "
+                "Please check that the base network {}s are defined correctly in "
+                "the config file.".format(
+                    nodes_or_links,
+                    subject_id,
+                    project_name,
+                    nodes_or_links,
+                )
+            )
+            raise ValueError(msg)
+
+
+def _find_available_id(
+    nodes_or_links: str,
+    project_name: str,
+    subject_id: str,
+    range_in_use: dict,
+    subject_df: pd.DataFrame,
+) -> int:
+    """
+    If the node or link id is already in the registry and we need to find a new number, this method iterates up from
+    the proposed node number to find the next available id, which it returns.
+
+    Args:
+        nodes_or_links (str): "node" or "link", which is used in error message
+        project_name (str): project name, which is used in error message
+        subject_id (int): the proposed new node or link id number
+        range_in_use (dict): a dictionary defining the id range with a bool indicating if the id number is used in the base network
+        subject_df (pd.DataFrame): node or link registry dataframe
+
+    Returns:
+        int: available node or link id
+    """
+
+    number = subject_id
+    for i in range(subject_id, max(range_in_use.keys())):
+        if i not in subject_df["id"].values:
+            if range_in_use[i] == False:
+                number = i
+                break
+
```

Comment:
I understand this part (that you are iterating to look for the next available id). However, I am still not fully clear on the IDs in registry_config.yml, registry.csv, and project cards and if looking for the next available id in `range(subject_id, max(range_in_use.keys()))` is the way to go about it. 

I will schedule a call to discuss this in detail.